src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,158 @@
+ *     <BreadcrumbItem onClick={() => {}}>Item 2</BreadcrumbItem>,
+ *   ]}
+ *   aria-label='Primary'
+ * />
+ />
+ */
+export const Breadcrumbs = React.forwardRef(
+  (props: BreadcrumbsProps, ref: React.RefObject<HTMLElement>) => {
+    const {
+      items,
+      currentIndex = items.length - 1,
+      separator,
+      className,
+      ...rest
+    } = props;
+
+    useTheme();
+
+    const breadcrumbsRef = React.useRef<HTMLElement>(null);
+
+    const [hiddenItems, setHiddenItems] = React.useState<JSX.Element[]>([]);
+    const [postItems, setPostItems] = React.useState(() => items.slice(1));
+    const overflowBreakpoints = React.useRef<number[]>([]);
+
+    const [breadcrumbsWidth, setBreadcrumbsWith] = React.useState<number>(0);
+    const updateBreadcrumbsWidth = React.useCallback(
+      ({ width }) => setBreadcrumbsWith(width),
+      [],
+    );
+    const [resizeRef] = useResizeObserver(updateBreadcrumbsWidth);
+    const refs = useMergedRefs(breadcrumbsRef, resizeRef, ref);
+
+    React.useLayoutEffect(() => {
+      if (!breadcrumbsRef.current) {
+        return;
+      }
+
+      // hide items when there's no space available
+      if (
+        breadcrumbsRef.current.offsetWidth < breadcrumbsRef.current.scrollWidth
+      ) {
+        if (postItems.length > 0) {
+          setHiddenItems([...hiddenItems, postItems.shift()!]); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+          setPostItems(postItems);
+          overflowBreakpoints.current.push(breadcrumbsWidth);
+        }
+      }
+      // restore items when there's enough space again
+      else if (
+        overflowBreakpoints.current.length > 0 &&
+        breadcrumbsWidth >
+          overflowBreakpoints.current[overflowBreakpoints.current.length - 1]
+      ) {
+        if (hiddenItems.length > 0) {
+          setPostItems([hiddenItems.pop()!, ...postItems]); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+          setHiddenItems([...hiddenItems]);
+          overflowBreakpoints.current.pop();
+        }
+      }
+    }, [breadcrumbsWidth, items, hiddenItems, postItems]);
+
+    const Separator = () => (
+      <li className='iui-breadcrumbs-separator' aria-hidden>
+        {separator ?? <SvgChevronRight />}
+      </li>
+    );
+
+    const ListItem = ({
+      item,
+      index,
+    }: {
+      item: JSX.Element;
+      index: number;
+    }) => (
+      <li
+        className={cx('iui-breadcrumbs-item', {
+          'iui-current': currentIndex === index,
+        })}
+      >
+        {React.cloneElement(item, {
+          'aria-current':
+            item.props['aria-current'] ?? currentIndex === index
+              ? 'location'
+              : undefined,
+        })}
+      </li>
+    );
+
+    return (
+      <nav className={cx('iui-breadcrumbs', className)} ref={refs} {...rest}>
+        <ol className='iui-breadcrumbs-list'>
+          <ListItem item={items[0]} index={0} />
+          <Separator />
```

Comment:
If there is just one item, then separator will be still shown.

Reply:
Fixed

Reply:
Really? 😃
![image](https://user-images.githubusercontent.com/36186912/126997491-743f3e2f-eec9-43ad-96d4-977b8c4931f0.png)

Reply:
There was a calculation error which I just fixed. Pull latest